extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -151,6 +152,9 @@ public Uni<SecurityIdentity> apply(HttpAuthenticationMechanism mech) {
 
                     });
         }
+        if (routingContext.get(ROLES_MAPPING_KEY) != null) {
```

Comment:
It does not look like that the the role related augmentation belongs here, it is indeed part of the authorization layer as you had it done originally.

Reply:
Why is `SecurityIdentityAugmentor` part of the authentication and `roles-mapping` augmentation not? If you mean because it differs between management/http app endpoints, then so does authentication mechanism selection (few lines above). I don't understand the difference. This feels like a right place.

Reply:
I mean, this is not done by any HTTP Security policy.

Reply:
@sberyozkin I was hinting at this line https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java#L128 which in my eyes is same situation. Authentication determined per management/application-endpoint.

Reply:
> It does not look like that the the role related augmentation belongs here, it is indeed part of the authorization layer as you had it done originally.

If possible @sberyozkin , please try to elaborate on this conclusion. I'm not able to find one concept how this should work in Quarkus Security. Yesterday you wrote me that authentication event should be fired when the authentication is done. In my eyes augmentation where it is not connected to authorization must be part of the authentication, therefore this move makes sense to me and solves the issue that no event contain augmented identity. Thank you very much for your thoughts and time.

Reply:
@michalvavrik Well, that link is indeed about a path specific authenticator. But this code is about the authorization related info. In any case, I'm not sure why this move is relevant for this PR ? Again, sorry if I did not get it from our earlier discussion

Reply:
> But this code is about the authorization related info.

The `quarkus.http.auth.roles-mapping.root=admin` remaps the identity roles. It is not connected to any authorization. Here is current implementation: https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/RolesMapping.java

I am probably opening Pandora box and you will ask me to re-implement the feautre :-D

> In any case, I'm not sure why this move is relevant for this PR ?

Because thanks to this move, roles will be remapped when (one of):

1. proactive authentication performs authentication
2. anyone subscribes to deferred SecurityIdentity

It means that it happens during the authentication. That's why when `AuthenticationSuccessEvent` is fired, then roles are already remapped. I need to report the identity with remapper roles via some event.

> why is it relevant?

because right now you can have `quarkus.http.auth.roles-mapping.root=admin` but no authorization, that is why this roles mapping is never reported.

> Again, sorry if I did not get it from our earlier discussion

No worry, I am happy to keep your attention on this PR. I don't know how else to fix OTel.